### PR TITLE
[DNM] Bare Metal

### PR DESCRIPTION
**Pull request by hzoppetti**
start of bare metal work

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -2987,6 +2989,16 @@ paths:
 
         **Important**: You must be an unrestricted User in order to add or modify
         tags on Linodes.
+
+        **Bare Metal Beta**
+        Creates a Bare Metal Instance on your account. In order for this request to complete successfully,
+        you must have the `add_metal` grants.
```
Comment (Dorthu) [inline openapi.yaml]: These are Linodes, the normal `add_linodes` grant is all you need.

File: openapi.yaml [hunk 2]
```diff
@@ -3304,6 +3326,15 @@ paths:
...
             within the billing period the Linode was active.
+
+        **Bare Metal Beta**
+        Deprovisions a Bare Metal Instance you have permissions to `read_write`.
+
+        Additionally, deleting a Bare Metal Instance:
+
+          * Gives up any IP addresses the Bare Metal Instance was assigned.
+          * Stops billing for the Bare Metal Instance and its associated services. You will be billed for time used
+            within the billing period the Bare Metal Instance was active.
```
Comment (Dorthu) [inline openapi.yaml]: None of this is different than just deleting a normal Linode.

File: openapi.yaml [hunk 3]
```diff
@@ -3667,6 +3710,9 @@ paths:
         * If there is more than one Config profile, the last booted config will be used.
         * If there is more than one Config profile and none were the last to be booted (because the
           Linode was never booted or the last booted config was deleted) an error will be returned.
+
+        **Bare Metal Beta**
+        Boots a Linode identified by the ID of the Bare Metal Linode you have permission to boot.
```
Comment (Dorthu) [inline openapi.yaml]: This is also the same as normal Linodes; the only difference for boot/reboot is that Config IDs are not accepted (are Bare Metal Instances don't have configs).

File: openapi.yaml [hunk 4]
```diff
@@ -5159,9 +5233,12 @@ paths:
     post:
       x-linode-grant: read_write
       summary: Reboot Linode
-      description: >
+      description: |
         Reboots a Linode you have permission to modify. If any actions are currently running or
         queued, those actions must be completed first before you can initiate a reboot.
+
+        **Bare Metal Beta**
+        Reboots a Bare Metal Instance you have permission to modify.
```
Comment (Dorthu) [inline openapi.yaml]: As above, this is the same as a normal Linode reboot, but you don't have the option to send in a `config_id`.

File: openapi.yaml [hunk 5]
```diff
@@ -5448,9 +5525,12 @@ paths:
     post:
       x-linode-grant: read_write
       summary: Shut Down Linode
-      description: >
+      description: |
         Shuts down a Linode you have permission to modify. If any actions are currently running or
         queued, those actions must be completed first before you can initiate a shutdown.
+
+        **Bare Metal Beta**
+        Shuts down a Bare Metal Instance you have permission to modify.
```
Comment (Dorthu) [inline openapi.yaml]: This is the same as a normal Linode.

**Comment by jamesottinger** [comment]
Replaced by #574

**Pull request closed without merging**